examples/metrics/grafana-dashboards/strimzi-cruise-control.json

```diff
@@ -0,0 +1,2062 @@
+          "text": "N/A",
+          "value": "null"
+        }
+      ],
+      "valueName": "current"
+    },
+    {
+      "gridPos": {
+        "h": 1,
+        "w": 24,
+        "x": 0,
+        "y": 4
+      },
+      "id": 114,
+      "title": "Cruise Control",
+      "type": "row"
+    },
+    {
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": "${DS_PROMETHEUS}",
+      "description": "Goal Violation Rate",
+      "fill": 1,
+      "fillGradient": 0,
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 5
+      },
+      "id": 106,
+      "legend": {
+        "alignAsTable": false,
+        "avg": false,
+        "current": false,
+        "hideEmpty": false,
+        "hideZero": false,
+        "max": false,
+        "min": false,
+        "rightSide": false,
+        "show": true,
+        "total": false,
+        "values": false
+      },
+      "lines": true,
+      "linewidth": 1,
+      "links": [],
+      "nullPointMode": "null",
+      "options": {
+        "dataLinks": []
+      },
+      "paceLength": 10,
+      "percentage": false,
+      "pointradius": 5,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "sum(irate(kafka_cruisecontrol_anomalydetector_goal_violation_rate_count{namespace=\"$kubernetes_namespace\",strimzi_io_cluster=\"$strimzi_cluster_name\",kubernetes_pod_name=~\"$strimzi_cluster_name-cruise-control-.*\"}[1m]))",
+          "format": "time_series",
+          "hide": false,
+          "instant": false,
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "Goal Violation Rate",
+          "refId": "A"
+        }
+      ],
+      "thresholds": [],
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Goal Violation Rate",
+      "tooltip": {
+        "shared": true,
+        "sort": 0,
+        "value_type": "individual"
+      },
+      "type": "graph",
+      "xaxis": {
+        "buckets": null,
+        "mode": "time",
+        "name": null,
+        "show": true,
+        "values": []
+      },
+      "yaxes": [
+        {
+          "decimals": null,
+          "format": "short",
+          "label": "violations/sec",
+          "logBase": 1,
+          "max": null,
+          "min": null,
+          "show": true
+        },
+        {
+          "format": "short",
+          "label": null,
+          "logBase": 1,
+          "max": null,
+          "min": null,
+          "show": true
+        }
+      ],
+      "yaxis": {
+        "align": false,
+        "alignLevel": null
+      }
+    },
+    {
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": "${DS_PROMETHEUS}",
+      "description": "Disk Failure Rate",
```

Comment:
What does this mean? That the disk died?

Reply:
So as far as I understood from the DiskFailureDetector code, it periodically tries to get info on the log dirs on the brokers (using the Admin Client API `describeLogDirs`) and the callback is able to provide info about errors which are potential disk failures.

Reply:
I think this is a bit scary then ... what about calling it _Failed disk queries_ or something like that?

Reply:
Maybe "disk read failure rate"?

Reply:
TBH, I think _Failed disk queries_ would be better because IMHO it makes it more clear that this is just about Cruise Control issue when querying the disk. _Disk read failure rate_ can still be easily confused with a physical disk issue. But I can live with it if needed.

Reply:
I see your point you want to highlight that it's something more related to CC than the disk itself even if when CC has a "query" failure, it could mean a real disk failure. @tomncooper any opinion on this?

Reply:
So the disk failures are detected using the code [here](https://github.com/linkedin/cruise-control/blob/a7ff697af10f20c881239dfe41283460965edbaf/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/DiskFailureDetector.java#L89). It seems to use the Kafka Admin client to ask for broker status and then checks for reported errors in each of the log directories.

Reply:
So this looks to be a replication of the Kafka side metrics?

Reply:
That was the code I was referring to (without linking the code actually). But what's a good name/description for you?

Reply:
Sorry @ppatierno I didn't read your comment at the top and basically repeated exactly what you said. Not very helpful. Anyway this metric is the number of disc failures in the last detection interval (which will be the same as the AD interval unless the user specifies a different interval for the disk failure detection), so it is a "rate" and it is a failure according to Kafka. So "Disk Failure Rate"/"Reported Disk Failure Rate" seems pretty accurate to me.